compiler/rustc_codegen_llvm/src/context.rs

```diff
@@ -585,7 +590,13 @@ impl<'ll, 'tcx> MiscMethods<'tcx> for CodegenCx<'ll, 'tcx> {
 
     fn declare_c_main(&self, fn_type: Self::Type) -> Option<Self::Function> {
         if self.get_declared_value("main").is_none() {
-            Some(self.declare_cfn("main", llvm::UnnamedAddr::Global, fn_type))
+            let visibility = if self.sess().target.default_hidden_visibility {
+                llvm::Visibility::Hidden
+            } else {
+                llvm::Visibility::Default
+            };
```

Comment:
Given the implication of `default_hidden_visibility` why is it that we’re specially handling just the `main` rather than applying this universally to all calls of `declare_cfn`?

My intuition is that intrinsics probably don’t care too much which way the visibility is set for them, which besides the exception stuff is the only other thing that utilizes `declare_cfn`.

Reply:
I was trying to avoid modifying the intrinsics, but it appears you're right that LLVM still accepts them with hidden visibility. I've now changed the patch to handle this universally in all calls of `declare_cfn`.